apps/website/src/pages/stream/overlay.tsx

```diff
@@ -78,20 +78,34 @@ const OverlayPage: NextPage = () => {
     return () => clearInterval(weatherInterval.current);
   }, []);
 
-  // Get the events for the next three days
+  // Set the range for upcoming events
   // Refresh every 60s
   const [upcomingRange, setUpcomingRange] = useState<[Date, Date]>();
+  const upcomingRangeInterval = useRef<NodeJS.Timeout>();
   useEffect(() => {
-    const now = new Date();
-    const next = new Date(now);
-    next.setDate(next.getDate() + 3);
-    setUpcomingRange([now, next]);
+    const updateRange = () => {
+      const now = new Date();
+      const next = new Date(now);
+      next.setDate(next.getDate() + 3);
+      setUpcomingRange([now, next]);
+    };
+
+    updateRange();
+    upcomingRangeInterval.current = setInterval(updateRange, 60 * 1000);
+    return () => clearInterval(upcomingRangeInterval.current);
   }, []);
-  const events = trpc.calendarEvents.getCalendarEvents.useQuery(
-    { start: upcomingRange?.[0], end: upcomingRange?.[1] },
-    { enabled: upcomingRange !== undefined, refetchInterval: 60 * 1000 },
-  );
-  const firstEventId = events.data?.[0]?.id;
+
+  // Get the upcoming events
+  // Refresh when the range changes
+  const { data: events, refetch } =
+    trpc.calendarEvents.getCalendarEvents.useQuery(
+      { start: upcomingRange?.[0], end: upcomingRange?.[1] },
+      { enabled: false },
+    );
+  useEffect(() => {
+    if (upcomingRange) refetch();
+  }, [upcomingRange, refetch]);
```

Comment:
Any reason we need the manual refetch? The state change should be enough to trigger the query.

```suggestion
  const { data: events } = trpc.calendarEvents.getCalendarEvents.useQuery(
    { start: upcomingRange?.[0], end: upcomingRange?.[1] },
    { enabled: upcomingRange !== undefined },
  );
```

Reply:
Doing it like this was doing all sorts of random refetches in the background, not just when the timestamps changed. I could not understand why.

Reply:
I see yeah it will refetch on window focus, which would happen a lot in dev ;) Updated the suggestion above, but doing it in the effect works too :)

Reply:
This still doesn't work though, `events` seems to become empty for a second as it gets the new data, which doesn't seem to happen with a manual refetch 🤔

Reply:
```ts
  const { data: events, isSuccess } =
    trpc.calendarEvents.getCalendarEvents.useQuery(
      { start: upcomingRange?.[0], end: upcomingRange?.[1] },
      { enabled: upcomingRange !== undefined },
    );
  const firstEventId = useMemo((prev?: string) => isSuccess
    ? events?.[0]?.id
    : prev, [isSuccess, events]);
```
I can do this instead if you think this is clearer (I'm thinking it might be)?